api/pkg/apis/v1alpha1/vendors/stage-vendor.go

```diff
@@ -248,12 +248,9 @@ func (s *StageVendor) Init(config vendors.VendorConfig, factories []managers.IMa
 		}
 
 		// restore schedule
-		var schedule *v1alpha2.ScheduleSpec
+		var schedule = ""
 		if v, ok := dataPackage.Inputs["__schedule"]; ok {
-			err = json.Unmarshal([]byte(v.(string)), &schedule)
-			if err != nil {
-				return err
-			}
+			schedule = v.(string)
```

Comment:
Prevent convert type directly which may cause process to crash. Either use utils.FormatAsString or schedule, ok = v.(string) and do error handling

Reply:
Looks like below code also use a same way....

		triggerData := v1alpha2.ActivationData{
			Activation:           dataPackage.Inputs["__activation"].(string),
			ActivationGeneration: dataPackage.Inputs["__activationGeneration"].(string),
			Campaign:             dataPackage.Inputs["__campaign"].(string),
			Stage:                dataPackage.Inputs["__stage"].(string),
			Inputs:               dataPackage.Inputs,
			Outputs:              dataPackage.Outputs,
			Schedule:             schedule,
			NeedsReport:          true,
			Namespace:            dataPackage.Inputs["__namespace"].(string),
		}

Reply:
fixed!